thread/common/threadhelpers.cpp

```diff
@@ -130,10 +136,11 @@ omrthread_spinlock_acquire(omrthread_t self, omrthread_monitor_t monitor)
 #endif /* OMR_THR_JLM */
 
 #if defined(OMR_THR_SPIN_WAKE_CONTROL)
-	VM_AtomicSupport::subtract(&monitor->spinThreads, 1);
-
-exit:
+	if (spinning && (OMRTHREAD_IGNORE_SPIN_THREAD_BOUND != lib->maxSpinThreads)) {
```

Comment:
I do not think this is correct, which could be causing monitor->spinThreads to be inflated.

Reply:
`(OMRTHREAD_IGNORE_SPIN_THREAD_BOUND != lib->maxSpinThreads)` - this should stop atomic increment/decrement of `monitor->spinThreads` and stop the (`monitor->spinThreads < lib->maxSpinThreads`) check to allow unlimited spin threads.